microsoftdynamicscrm/metadata.go

```diff
@@ -2,13 +2,125 @@ package microsoftdynamicscrm
 
 import (
 	"context"
+	"errors"
+	"fmt"
 
 	"github.com/amp-labs/connectors/common"
+	"github.com/subchen/go-xmldom"
+)
+
+var (
+	CRMMetadataSchemaName = "Microsoft.Dynamics.CRM" // nolint:gochecknoglobals
+
+	ErrMissingSchema      = fmt.Errorf("missing schema %v in response", CRMMetadataSchemaName)
+	ErrMetadataProcessing = errors.New("metadata couldn't be processed")
+	ErrObjectNotFound     = errors.New("object not found")
 )
 
 func (c *Connector) ListObjectMetadata(
 	ctx context.Context, objectNames []string,
 ) (*common.ListObjectMetadataResult, error) {
-	// TODO implement me
-	panic("implement me")
+	// Ensure that objectNames is not empty
+	if len(objectNames) == 0 {
+		return nil, common.ErrMissingObjects
+	}
+
+	rsp, err := c.getXML(ctx, c.getURL("$metadata"))
+	if err != nil {
+		return nil, err
+	}
+
+	root, err := rsp.GetRoot()
+	if err != nil {
+		return nil, err
+	}
+
+	entities, err := extractEntities(root)
+	if err != nil {
+		return nil, err
+	}
+	// FIXME a call to the metadata endpoint is costly, once queried and processed we should Cache entities
```

Comment:
Can you remove this FIXME? this is a concern of the server, this library doesn't do any caching.